util/iputil/matcher.go

```diff
@@ -0,0 +1,48 @@
+package iputil
+
+import (
+	"net"
+)
+
+// IPMatcher is the interface for matching on an ip address and version.
+type IPMatcher interface {
```

Comment:
I wonder if we should change "Match" to "Validate". We have the counterintuitive result that if `Match()` returns true, the IP was not on the list of IPs in the IPMatcher

Reply:
Yeah. I cringed at that too. We're looking for public networks from a list of private networks, so there's got to be a negation somewhere. Are you suggesting only a name change or to flip the logic too?

I choose the matcher name by following the naming pattern used by the standard sort package. I'm sure having FindIP take an IPValidator works too. :)

Reply:
I was just thinking of changing the name. If we flip the logic, then there is no need to change the name in my view.

Reply:
Sure.

Reply:
Renamed to IPValidator and IPValidator.IsValid.